stream_alert/shared/description.py

```diff
@@ -67,22 +82,31 @@ def parse(cls, rule_description):
             dict: A dict mapping fields to lists of strings, each corresponding to a line belonging
                   to that field. All field names are lowercase.
         """
-        tokens = rule_description.split('\n')
+        rule_description = '' if not rule_description else rule_description
+        tokens = [line.strip() for line in rule_description.strip().split('\n')]
 
         field_lines = {}
 
         current_field = 'description'
         for token in tokens:
-            if not token or not token.strip():
+            if current_field not in field_lines:
+                field_lines[current_field] = []
+
+            if not token:
+                field_lines[current_field].append('')
```

Comment:
Why do we append an empty string to a list?

Reply:
I added a comment to make it clearer why; we need to preserve empty lines so that we know how to recognize when there is a true (markdown-style) line break